oops/backplane/unittester_support.py

```diff
@@ -0,0 +1,503 @@
+def _save_image(image, filename):
+    """
+    Read an image file of a 2-D array.
+
+    Input:
+        image       a 2-D byte array.
+        filename    the name of the output file, which should end with the
+                    type, e.g., '.png' or '.jpg'
+    """
+
+    shape = image.shape[::-1]
+    if len(shape) == 1:
+        shape = (shape[0],1)
+
+    im = Image.frombytes('L', shape, image)
+    im.save(filename)
+
+
+
+#===============================================================================
+def _scale_image(array, minval, maxval):
+    """Rescales an image and converts to byte."""
+
+    image = array.vals.copy()
+    image[array.mask] = minval - 0.05 * (maxval - minval)
+
+    image = np.asfarray(image)
+
+    lo = image.min()
+    hi = image.max()
+
+    if hi == lo:
+        bytes = np.zeros(image.shape, dtype=np.int8)
+    else:
+        scaled = (image[::-1] - lo) / float(hi - lo)
+        bytes = (256.*scaled).clip(0,255).astype('uint8')
+
+    return bytes
+
+
+
+#===============================================================================
+def _compare_backplanes(array, reference, margin=0.05):
+    """Compare a backplane array to the reference array."""
+
+    array = Scalar.as_scalar(array)
+    reference = Scalar.as_scalar(reference)
+
+    diff = abs(array - reference)
+    assert diff.max() <= reference.max()*margin
+
+
+
+#===============================================================================
+def _convert_filename(filename):
+    """Converts file-system-unfriendly characters in a backplane filename."""
+
+    filename = filename.replace(':','_')
+    filename = filename.replace('/','_')
+    filename = filename.replace(' ','_')
+    filename = filename.replace('(','_')
+    filename = filename.replace(')','_')
+    filename = filename.replace('[','_')
+    filename = filename.replace(']','_')
+    filename = filename.replace('&','_')
+    filename = filename.replace(',','_')
+    filename = filename.replace('-','_')
+    filename = filename.replace('__','_')
+    filename = filename.replace('__','_')
+    filename = filename.replace('_.','.')
+    filename = filename.replace("'",'')
+    return filename.lower().rstrip('_')
+
+
+
+#===============================================================================
+def _construct_filename(bp, array, title, dir):
+    """Constructs a backplane filename."""
+
+    # Construct base filename from title
+    filename = _convert_filename('backplane-' + \
+                                 bp.obs.basename.split('.')[0] + '-' + title)
+
+    # Ensure unique filename by using the backplane key, if it exists
+    # NOTE: if no backplane key exists, a non-unique filename could result
+    try:
+        key = array.key
+        id = key[0]
+        for item in key[1:]:
+            if item != ():
+                id = id + '_' + str(item)
+        id = _convert_filename(id)
+        filename = filename + '[' + id + ']' + '.png'
+    except:
```

Comment:
Bad naked except. Also too much code inside the try. Lots of chance for false exceptions.